src/main/java/org/zalando/nakadi/service/job/TimelineCleanupJob.java

```diff
@@ -0,0 +1,81 @@
+import org.springframework.stereotype.Service;
+import org.zalando.nakadi.domain.Timeline;
+import org.zalando.nakadi.exceptions.TopicDeletionException;
+import org.zalando.nakadi.exceptions.runtime.InconsistentStateException;
+import org.zalando.nakadi.exceptions.runtime.RepositoryProblemException;
+import org.zalando.nakadi.repository.TopicRepository;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+import org.zalando.nakadi.repository.db.TimelineDbRepository;
+import org.zalando.nakadi.service.timeline.TimelineService;
+
+@Service
+public class TimelineCleanupJob {
+
+    private static final String JOB_NAME = "timelines-cleanup";
+
+    private static final Logger LOG = LoggerFactory.getLogger(TimelineCleanupJob.class);
+
+    private final EventTypeCache eventTypeCache;
+    private final TimelineDbRepository timelineDbRepository;
+    private final TimelineService timelineService;
+    private final ExclusiveJobWrapper jobWrapper;
+
+    @Autowired
+    public TimelineCleanupJob(final EventTypeCache eventTypeCache,
+                              final TimelineDbRepository timelineDbRepository,
+                              final TimelineService timelineService,
+                              final JobWrapperFactory jobWrapperFactory,
+                              @Value("${nakadi.jobs.timelineCleanup.runPeriodMs}") final int periodMs) {
+        this.eventTypeCache = eventTypeCache;
+        this.timelineDbRepository = timelineDbRepository;
+        this.timelineService = timelineService;
+        this.jobWrapper = jobWrapperFactory.createExclusiveJobWrapper(JOB_NAME, periodMs);
+    }
+
+    @Scheduled(
+            fixedDelayString = "${nakadi.jobs.checkRunMs}",
+            initialDelayString = "${random.int(${nakadi.jobs.checkRunMs})}")
+    public void cleanupTimelines() {
+        try {
+            jobWrapper.runJobLocked(() ->
+                    timelineDbRepository.getExpiredTimelines().stream()
+                            .forEach(timeline -> {
+                                deleteTimelineTopic(timeline);
+                                markTimelineDeleted(timeline);
+                            }));
+        } catch (final RepositoryProblemException e) {
+            LOG.error("DB error occurred when trying to get expired timelines", e);
+        }
+    }
+
+    private void deleteTimelineTopic(final Timeline timeline) {
+        try {
+            final TopicRepository topicRepository = timelineService.getTopicRepository(timeline);
+            topicRepository.deleteTopic(timeline.getTopic());
+        } catch (final TopicDeletionException e) {
+            LOG.error("Failed to delete topic {} for expired timeline {}", timeline.getTopic(), timeline.getId(), e);
+        }
+    }
+
+    private void markTimelineDeleted(final Timeline timeline) {
+        try {
+            timeline.setDeleted(true);
+            timelineDbRepository.updateTimelime(timeline);
```

Comment:
Ok, I added this functionality https://github.com/zalando/nakadi/pull/611/commits/69d1e9ed5a3589d65e93ff00e1a6253ba65a1adc